### PR TITLE
feat(shared-cache): Respect CacheItemRequest::should_load

### PR DESCRIPTION
**Pull request by flub**
Given that the cache system has this it should probably be respected.

NATIVE-248

#skip-changelog

### CODE REVIEW

**Review by loewenheim — APPROVED** [review]

File: crates/symbolicator/src/services/cacher.rs [hunk 1]
```diff
@@ -308,26 +308,39 @@ impl<T: CacheItemRequest> Cacher<T> {
...
+            metric!(counter("shared_cache.file.discarded") += 1);
+        }
+
+        let status = match status {
+            Some(status) => status,
+            None => {
+                let status = request.compute(temp_file.path()).await?;
+                status.write(&mut temp_fd).await?;
+                status
+            }
```
Comment (loewenheim) [inline crates/symbolicator/src/services/cacher.rs:337]: Nit: I think this is just `status.unwrap_or_else(|| …)`, but it's fine like this as well.

Comment (flub) [reply crates/symbolicator/src/services/cacher.rs:337]: you need `.await` though, which you can't write in an `.unwrap_or_else()`

Comment (loewenheim) [reply crates/symbolicator/src/services/cacher.rs:337]: Thanks, I missed that.

**Review by Swatinem — APPROVED** [review]

File: crates/symbolicator/src/services/cacher.rs [hunk 2]
```diff
@@ -308,26 +308,39 @@ impl<T: CacheItemRequest> Cacher<T> {
...
             .await;
 
-        let status = if !shared_cache_hit {
-            let status = request.compute(temp_file.path()).await?;
-            status.write(&mut temp_fd).await?;
-            Some(status)
-        } else {
-            None
+        let status = match shared_cache_hit {
+            true => {
```
Comment (Swatinem) [inline crates/symbolicator/src/services/cacher.rs:317]: match on true/false is super weird, I wonder why clippy does not complain, lol

Comment (flub) [reply crates/symbolicator/src/services/cacher.rs:317]: heh, just style i think, i generally prefer match over if...else.  doubly so when assigning the result of the expression.

**Pull request merged**